src/ucp/core/ucp_worker.c

```diff
@@ -2388,6 +2438,15 @@ static void ucp_worker_destroy_eps(ucp_worker_h worker)
                            UCS_STATUS_PTR(UCS_ERR_CANCELED));
         ucp_ep_disconnected(ep, 1);
     }
+
+    /* Destroying UCP EP in ucp_ep_disconnected() could start UCT EP discarding
+     * operations (e.g. for TMP EP in CM lane) which should be completed here
+     * in order to destroy all UCP EPs */
+    ucp_worker_discard_uct_ep_purge_reqs(worker);
```

Comment:
logically, it should no be part of ucp_worker_destroy_eps

Reply:
yes

Reply:
done